app/controllers/admin_controller.rb

```diff
@@ -1,7 +1,49 @@
 class AdminController < ApplicationController
+  before_action :authorize_admin
 
   def index
     @membership_applications = MembershipApplication.all
   end
 
+
+  # export membership_appplications
+  def export_ansokan_csv
+
+    @membership_applications = MembershipApplication.all
+
+    download_dir = File.join(Rails.root, 'tmp', 'downloads')
+    Dir.mkdir download_dir unless Dir.exist? download_dir
+
+    export_name = "shf-ankosan-#{Time.now.strftime('%Y-%m-%d--%H-%M-%S')}.csv"
+
+    send_data( export_str(@membership_applications), filename: export_name, type: "text/plain")
+
+    helpers.flash_message(:notice, t('.success'))
+
```

Comment:
Suggest adding a rescue clause *just in case*, something simple like
```
rescue RuntimeError => e
    flash[:alert] = "Error: #{e}"
    redirect_to (request.referer.present? ? :back : root_path)
```